File: taloa-api/app/schemas/travel.py
"""Schemas do Travel Checklist (Etapa 28).

REGRA: nada disto e publico. So o dono autenticado (ou admin), com assinatura
ativa (Plus+), validado server-side por ownership + require_active_subscription.
"""
from datetime import date
from typing import Literal

from pydantic import BaseModel, Field

TravelType = Literal["car", "plane", "ferry", "train"]
TripScope = Literal["domestic", "international"]
ItemSection = Literal["documents", "transport", "essentials"]


class TripCreate(BaseModel):
    title: str | None = Field(default=None, max_length=200)
    travel_type: TravelType
    scope: TripScope
    destination: str | None = Field(default=None, max_length=200)
    travel_date: date


class ChecklistItem(BaseModel):
    id: str
    item_key: str | None = None   # template; None = custom
    label: str | None = None      # so custom
    section: str
    is_checked: bool = False
    due_date: str | None = None
    sort_order: int = 0


class TripSummary(BaseModel):
    id: str
    title: str | None = None
    travel_type: str
    scope: str
    destination: str | None = None
    travel_date: str
    checked_count: int = 0
    total_count: int = 0
    created_at: str | None = None


class TripDetail(TripSummary):
    items: list[ChecklistItem] = []


class CustomItemCreate(BaseModel):
    label: str = Field(min_length=1, max_length=200)


class ItemUpdate(BaseModel):
    is_checked: bool

File: taloa-web/types/travel.ts
// Tipos do Travel Checklist (Etapa 28). Espelham schemas/travel.py da API.
export type TravelType = "car" | "plane" | "ferry" | "train";
export type TripScope = "domestic" | "international";
export type ItemSection = "documents" | "transport" | "essentials";

export interface ChecklistItem {
  id: string;
  item_key: string | null; // template (i18n); null = custom
  label: string | null;    // so custom
  section: ItemSection;
  is_checked: boolean;
  due_date: string | null; // ISO date
  sort_order: number;
}

export interface TripSummary {
  id: string;
  title: string | null;
  travel_type: TravelType;
  scope: TripScope;
  destination: string | null;
  travel_date: string;
  checked_count: number;
  total_count: number;
  created_at: string | null;
}

export interface TripDetail extends TripSummary {
  items: ChecklistItem[];
}

export interface TripPayload {
  title?: string;
  travel_type: TravelType;
  scope: TripScope;
  destination?: string;
  travel_date: string;
}
